go/types/meta_sequence.go

```diff
@@ -383,27 +383,24 @@ func (ms metaSequence) getChildren(start, end uint64) (seqs []sequence) {
 	d.Chk.True(start <= end)
 
 	seqs = make([]sequence, end-start)
-	hs := make(hash.HashSet, len(seqs))
+	hs := make(hash.HashSlice, len(seqs))
 
 	dec := ms.decoder()
 
 	for i := start; i < end; i++ {
-		hs[ms.getRefAt(&dec, int(i)).TargetHash()] = struct{}{}
+		hs[i-start] = ms.getRefAt(&dec, int(i)).TargetHash()
 	}
 
 	if len(hs) == 0 {
 		return // can occur with ptree that is fully uncommitted
 	}
 
 	// Fetch committed child sequences in a single batch
-	valueChan := make(chan Value, len(hs))
-	go func() {
-		ms.vrw.ReadManyValues(hs, valueChan)
-		close(valueChan)
-	}()
+	readValues := ms.vrw.ReadManyValues(hs)
+
 	children := make(map[hash.Hash]sequence, len(hs))
```

Comment:
same thing. you can ditch `children` here since it was just serving as a mapping from hash -> position (maybe a bit of math required as well).